QueryBuilder.Test/QueryBuilder.Dynamic/Where.UnitTests.cs

```diff
@@ -0,0 +1,452 @@
+        }
+
+        [TestMethod]
+        public void OperatorsHaveTheRightNames()
+        {
+            Assert.AreEqual("=", ComparisonOperators.IsEqualTo.Operator);
+            Assert.AreEqual(">", ComparisonOperators.IsGreaterThan.Operator);
+            Assert.AreEqual(">=", ComparisonOperators.IsGreaterThanOrEqualTo.Operator);
+            Assert.AreEqual("<", ComparisonOperators.IsLessThan.Operator);
+            Assert.AreEqual("<=", ComparisonOperators.IsLessThanOrEqualTo.Operator);
+            Assert.AreEqual("!=", ComparisonOperators.NotEqualTo.Operator);
+            Assert.AreEqual("IS_BOOL", ScalarOperators.IS_BOOL.Name);
+            Assert.AreEqual("IS_DEFINED", ScalarOperators.IS_DEFINED.Name);
+            Assert.AreEqual("IS_NULL", ScalarOperators.IS_NULL.Name);
+            Assert.AreEqual("IS_NUMBER", ScalarOperators.IS_NUMBER.Name);
+            Assert.AreEqual("IS_OBJECT", ScalarOperators.IS_OBJECT.Name);
+            Assert.AreEqual("IS_PRIMITIVE", ScalarOperators.IS_PRIMITIVE.Name);
+            Assert.AreEqual("IS_STRING", ScalarOperators.IS_STRING.Name);
+            Assert.AreEqual("STARTSWITH", ScalarOperators.STARTSWITH.Name);
+            Assert.AreEqual("ENDSWITH", ScalarOperators.ENDSWITH.Name);
+            Assert.AreEqual("CONTAINS", ScalarOperators.CONTAINS.Name);
+        }
+
+        [TestMethod]
+        public void CanUseEscapableCharacters()
+        {
+            var someBuilding1 = "some'building";
+            var someBuildingEscaped1 = @"some\'building";
+            var someBuilding2 = "some'buil'ding";
+            var someBuildingEscaped2 = @"some\'buil\'ding";
+
+            var query = DynamicQueryBuilder
+                      .FromTwins()
+                      .Where(b => b.Property("name").IsEqualTo(someBuilding1));
+
+            Assert.AreEqual($"SELECT twin FROM DIGITALTWINS twin WHERE twin.name = '{someBuildingEscaped1}'", query.BuildAdtQuery());
+
+            query = DynamicQueryBuilder
+                  .FromTwins()
+                  .Where(b => b.Property("name").IsIn(new string[] { someBuilding1, someBuilding2 }));
+
+            Assert.AreEqual($"SELECT twin FROM DIGITALTWINS twin WHERE twin.name IN ['{someBuildingEscaped1}','{someBuildingEscaped2}']", query.BuildAdtQuery());
+
+            query = DynamicQueryBuilder
+                  .FromTwins()
+                  .Where(b => b.Property("name").StartsWith(someBuilding1));
+
+            Assert.AreEqual($"SELECT twin FROM DIGITALTWINS twin WHERE STARTSWITH(twin.name, '{someBuildingEscaped1}')", query.BuildAdtQuery());
+        }
+
+        [TestMethod]
+        public void FilteringOverEnumWithNoMemberAttributeUsesInt()
+        {
+            var query = DynamicQueryBuilder
+                    .FromTwins()
+                    .Where(x => x.Property("status").IsEqualTo(CustomEnum.Value2));
+
+            Assert.AreEqual($"SELECT twin FROM DIGITALTWINS twin WHERE twin.status = '1'", query.BuildAdtQuery());
```

Comment:
If the enum is backed by an int, shouldn't the query write the int without quotes?

Reply:
Not sure, we haven't made a twin with a numeric enum that I recall. I just followed how we had it before. I'd have to test unless Amr wrote it this way because he had already checked.